### ISSUE TITLE
`connections/connection.py` failing non-responsive port test

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#3622]
The `connections/connection.py` test is currently failing with the following message:

```
FAIL: Test that we get a ReQL error if we connect to a non-responsive port
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/graham/wd/rethinkdb/test/rql_test/connections/connection.py", line 170, in test_connnect_timeout
    self.assertRaisesRegexp(r.RqlDriverError, "Timed out during handshake with localhost:%d." % port, r.connect, port=port, timeout=2)
  File "/Users/graham/wd/rethinkdb/test/rql_test/connections/connection.py", line 99, in replacement_assertRaisesRegexp
    self.assertTrue(re.search(regexp, str(e)), '%s did not raise the expected message "%s", but rather: %s' % (repr(callable_func), str(regexp), str(e)))
AssertionError: <function connect at 0x101bf37d0> did not raise the expected message "Timed out during handshake with localhost:64673.", but rather: Interrupted connection while receiving from localhost:64673 - timed out
```

The issue looks like the change in the return errors from `_sock_recvall` is not being caught and re-wrapped in line 182 of `/drivers/python/rethinkdb/net.py`.

**Comment by larkost** [comment]
This is up for review as CR 2505, from branch `larkost/3622-connection.py-no-reponsive-port`

**Comment by larkost** [comment]
This has been merged into `next` with f6056c5, and cherry-picked (with @danielmewes's ok) into `v1.16.x` with 98951be.
